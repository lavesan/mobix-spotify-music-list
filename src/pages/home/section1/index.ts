import SectionOneComponent from './section1.component';

export {
    SectionOneComponent,
}