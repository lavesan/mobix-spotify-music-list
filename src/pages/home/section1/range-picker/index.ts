import RangePickerComponent from './range-picker.component';

export {
    RangePickerComponent,
}