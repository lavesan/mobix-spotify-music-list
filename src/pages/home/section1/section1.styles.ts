import styled, { css } from 'styled-components';

export const StyledArtistsFilterBox = styled.section`
    display: grid;
    grid-template-columns: 3fr 4fr 7fr;
    grid-gap: 30px;
    margin-bottom: 30px;
    min-width: 0;
    min-height: 0;

    @media(max-width: 1100px) {
        grid-template-columns: 3fr 4fr 4fr;
    }

    @media(max-width: 880px) {
        grid-template-columns: 1fr 1fr;
    }

    @media(max-width: 600px) {
        grid-template-columns: 1fr;
    }
`;


export const StyledLogSpotifyLink = styled.a`
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 5px;
    text-decoration: none;
    width: 50%;
    margin: 0 auto;
    padding: 10px 20px;
    min-width: 0;

    ${({ theme }) => 
        css`
            background-color: ${theme.spotify.green.primaryColor};
            color: ${theme.spotify.dark.primaryColor};

            
            :hover {
                color: ${theme.spotify.dark.primaryColor};
                text-decoration: none;
            }
        `
    };
    
    @media(max-width: 1100px) {
        width: 80%;
    }

    @media(max-width: 880px) {
        width: 85%;
    }
`;