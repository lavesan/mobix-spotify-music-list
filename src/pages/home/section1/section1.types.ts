export type SelectArtistType = {
    value: string;
    label: string;
    color: string;
}