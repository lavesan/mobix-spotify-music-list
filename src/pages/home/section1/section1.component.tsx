import React, { useContext, useEffect } from 'react';
import { StyledH2 } from '../home.styles';
import { StyledArtistsFilterBox, StyledLogSpotifyLink } from './section1.styles';
import { SelectComponent } from './select';
import { RangePickerComponent } from './range-picker';
import { AppContext } from '../../../app/App.context';
import { HomePageContext } from '../home.context';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpotify } from '@fortawesome/free-brands-svg-icons';
import { MusicCardComponentProps } from '../section3/music-card/music-card.types';

export default () => {
    const { spotifyService } = useContext(AppContext);
    const { setChartData, setMusicsListed } = useContext(HomePageContext);

    const options = [
        { value: '1on7ZQ2pvgeQF4vmIA09x5', label: 'Charlie Brown Jr.' },
        { value: '14pVkFUHDL207LzLHtSA18', label: 'Pantera' },
        { value: '1dfeR4HaWDbWqFHLkxsg1d', label: 'Queen' },
        { value: '3lDpdwM8KILepMHqBWUhIA', label: 'The Struts' },
        { value: '2ZofT7n9AlTKf7KDCoHGgD', label: 'Luiz Gonzaga' },
    ];

    const handleChooseArtist = async ({ value }: any) => {
        const albums = await spotifyService.getArtistAlbums(value);
        if (albums) {
            setChartData(albums.items.map(({ id, name, release_date, total_tracks }: any) => ({
                id: id,
                name: name,
                releaseDate: release_date,
                totalTrack: total_tracks,
            })))
        }
    }

    useEffect(() => {
        // Gets all top tracks of all artists on the select option
        (async () => {
            let musicTracks: MusicCardComponentProps[] = [];
            for(const { value } of options) {
                const artistTopTrack = await spotifyService.getArtistTopTrack(value);
                if (artistTopTrack) {
                    musicTracks = [
                        ...musicTracks,
                        ...artistTopTrack.tracks.map(({ disc_number, explicit, name, popularity, track_number, album: { id } }: any) => ({
                            explicit,
                            popularity,
                            albumId: id,
                            diskNumber: disc_number,
                            trackNumber: track_number,
                            musicName: name,
                        }))
                    ]
                }
            }
            setMusicsListed(musicTracks);
        })()
    }, [spotifyService, options, setMusicsListed]);

    return (
        <>
            <StyledH2>FILTRO DOS ARTISTAS</StyledH2>
            <StyledArtistsFilterBox>
                <SelectComponent
                    options={options}
                    placeholder="Artistas"
                    onChange={handleChooseArtist} />
                <RangePickerComponent
                    label="Periodo"
                    initialValue={2019} />
                <StyledLogSpotifyLink
                    title="Autentique-se no spotify"
                    href={`https://accounts.spotify.com/pt-BR/authorize?client_id=${process.env.REACT_APP_SPOTIFY_CLIENT_ID}&response_type=token&redirect_uri=${process.env.REACT_APP_SPOTIFY_LOGIN_REDIRECT_URI}`}>
                    <FontAwesomeIcon icon={faSpotify} />&nbsp;&nbsp;<span>Logar-se no spotify</span>
                </StyledLogSpotifyLink>
            </StyledArtistsFilterBox>
        </>
    )
}