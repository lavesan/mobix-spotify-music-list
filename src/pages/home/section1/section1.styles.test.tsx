import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import 'jest-styled-components';
import { StyledArtistsFilterBox, StyledLogSpotifyLink } from './section1.styles';
import { ThemeProvider } from 'styled-components';
import theme from '../../../app/App.theme';

describe('Tests `StyledArtistsFilterBox` ', () => {
  let tree: ReactTestRendererJSON | null;

  beforeAll(() => {
    tree = renderer.create(<StyledArtistsFilterBox />).toJSON();
  })

  test('Matches snapshot ', () => {
    expect(tree).toMatchSnapshot();
  })
  
  test('Have styles ', () => {
    expect(tree).toHaveStyleRule('grid-template-columns', '3fr 4fr 7fr')
    expect(tree).toHaveStyleRule('grid-template-columns', '3fr 4fr 4fr', {
      media: '(max-width: 1100px)',
    })
    expect(tree).toHaveStyleRule('grid-template-columns', '1fr 1fr', {
      media: '(max-width: 880px)',
    })
    expect(tree).toHaveStyleRule('grid-template-columns', '1fr', {
      media: '(max-width: 600px)',
    })
  })
})

describe('Tests `StyledLogSpotifyLink` ', () => {
  let tree: ReactTestRendererJSON | null;

  beforeAll(() => {
    tree = renderer.create(
        <ThemeProvider theme={theme}>
            <StyledLogSpotifyLink />
        </ThemeProvider>
    ).toJSON();
  })

  test('Matches snapshot ', () => {
    expect(tree).toMatchSnapshot();
  })
  
  test('Have styles ', () => {
    expect(tree).toHaveStyleRule('width', '50%')
    expect(tree).toHaveStyleRule('width', '80%', {
      media: '(max-width: 1100px)',
    })
    expect(tree).toHaveStyleRule('width', '85%', {
      media: '(max-width: 880px)',
    })
  })
})
