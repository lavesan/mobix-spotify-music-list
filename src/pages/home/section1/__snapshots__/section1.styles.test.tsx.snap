// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests \`StyledArtistsFilterBox\`  Matches snapshot  1`] = `
.c0 {
  display: grid;
  grid-template-columns: 3fr 4fr 7fr;
  grid-gap: 30px;
  margin-bottom: 30px;
  min-width: 0;
  min-height: 0;
}

@media (max-width:1100px) {
  .c0 {
    grid-template-columns: 3fr 4fr 4fr;
  }
}

@media (max-width:880px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
  }
}

@media (max-width:600px) {
  .c0 {
    grid-template-columns: 1fr;
  }
}

<section
  className="c0"
/>
`;

exports[`Tests \`StyledLogSpotifyLink\`  Matches snapshot  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 50%;
  margin: 0 auto;
  padding: 10px 20px;
  min-width: 0;
  background-color: #1db954;
  color: #212121;
}

.c0:hover {
  color: #212121;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (max-width:1100px) {
  .c0 {
    width: 80%;
  }
}

@media (max-width:880px) {
  .c0 {
    width: 85%;
  }
}

<a
  className="c0"
/>
`;
